ml-fundamentals/ml-problem-formulation-envisioning.md

```diff
@@ -0,0 +1,115 @@
+
+The main goal of envisioning is to work alongside subject matter experts to formulate the problem in such a way that there is a shared understanding of the problem domain, a clear goal, and a predefined approach to evaluating a potential solution; this is often done through a series of 'envisioning' sessions.
+
+## Understanding the problem domain
+
+Generally, before defining a project scope for a data science investigation, we must first understand the problem domain:
+
+* What is the problem?
+* Why does the problem need to be solved?
+* Does this problem require a machine learning solution?
+* How would a potential solution be used?
+
+However, establishing this understanding can prove difficult, especially for those unfamiliar with the problem domain. To ease this process, we can approach problems in a structured way by taking the following steps:  
+
+* Identify a measurable problem and define this in business terms, such that there is a clear objective and a good understanding of the input elements that can be controlled which may affect this objective; here, it is best to be as specific as possible.
+* Decide how the performance of a solution should be measured, such that it is aligned with the business objective, and the data needed to evaluate the effectiveness of a solution. Note that the data required to evaluate a solution's impact on the overall objective is not necessarily the same as that which may be required to create a solution.
+* Thinking about the solution as a black box, detail the function that a solution to this problem should perform to fulfil the objective and verify that the relevant data is available to solve the problem.
+  * One way of approaching this is by thinking about how a subject matter expert could solve the problem manually and the data that would be required; if a human subject matter expert is unable to use the available data to solve the problem, this is indicative that additional information is required and/or more data needs to be collected.
+* Based on the available data, define specific hypothesis statements - which can be proved
+or disproved - to guide the exploration of the data science team. Where possible, each hypothesis statement should have a clearly defined success criteria (e.g., *with an accuracy of over 60%*), however, this is not always possible - especially for projects where no solution to the problem currently exists. In these cases, the measure of success could be based on a subject matter expert verifying that the results meet their expectations.
+* Document all the above information, to ensure alignment between stakeholders and establish a clear understanding of the problem to be solved. Try to ensure that as much relevant domain knowledge is captured as possible, and that the features present in available data - and the way that the data was collected - are clearly explained, such that they can be understood by a non-subject matter expert.
```

Comment:
Are there some good templates or tools to document this? maybe a definition of done, or some ADR like template?

Reply:
Not that I am aware of, that could be a good area of future work